packages/gatsby/src/utils/__tests__/merge-gatsby-config.js

```diff
@@ -0,0 +1,83 @@
+const mergeGatsbyConfig = require(`../merge-gatsby-config`)
+
+describe(`Merge gatsby config`, () => {
+  it(`Merging empty config is an identity operation`, () => {
+    const emptyConfig = {}
+    const basicConfig = {
+      plugins: [`gatsby-mdx`],
+    }
+
+    expect(mergeGatsbyConfig(basicConfig, emptyConfig)).toEqual(basicConfig)
+    expect(mergeGatsbyConfig(emptyConfig, basicConfig)).toEqual(basicConfig)
+  })
+
+  it(`Merging plugins concatenates them`, () => {
+    const basicConfig = {
+      plugins: [`gatsby-mdx`],
+    }
+    const morePlugins = {
+      plugins: [`a-plugin`, `b-plugin`, { resolve: `c-plugin`, options: {} }],
+    }
+    expect(mergeGatsbyConfig(basicConfig, morePlugins)).toEqual({
+      plugins: [
+        `gatsby-mdx`,
+        `a-plugin`,
+        `b-plugin`,
+        { resolve: `c-plugin`, options: {} },
```

Comment:
Would it make sense to add a test here for duplicates? That seems like something that will need to be handled very deliberately, so a test seems prudent.

Reply:
yes, I had left it off to deal with when thinking more about sub-theming but we can definitely add "filter out exact match duplicates" now without affecting anything I think. I'll add `_.uniqWith(plugins, _.isEqual)` for now (with a test), which should cover exact matches across strings and objects.